### PR TITLE
Add tests in Github Actions.

### PR DESCRIPTION
**Pull request by sbrodehl**
This PR adds tests in Github Actions, and replaces the Travis pipeline.

I don't know much about `nose`, but it has been in maintenance mode for the past several years and is missing support for python >=3.10, so I think the testing system needs an upgrade.

Edit: It also partly replaces the Appveyor pipeline, since the tests are run on MacOS, Ubuntu and Windows.

### CODE REVIEW

**Comment by sbrodehl** [comment]
Yes, you are right!
I removed `nose` as a dependency and used pytest instead.

This PR tests on MacOS, Ubuntu and Windows, and as noted in #55, it partly replaces the Appveyor pipeline.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+jobs:
+    packages:
+        name: Test on ${{ matrix.os }} (${{ matrix.python }})
+        runs-on: ${{ matrix.os }}
+        strategy:
+          fail-fast: true
+          matrix:
+            os: [ubuntu-latest, macos-latest, windows-latest]
+            python: ["3.6", "3.7", "3.8", "3.9", "3.x"]
```
Comment (fhs) [inline .github/workflows/tests.yml:18]: i think we should add 3.10 and remove 3.6, which is EOL

**Review by fhs — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (fhs) [reply .github/workflows/tests.yml:18]: oh, wait, is 3.x == 3.10?

Comment (sbrodehl) [reply .github/workflows/tests.yml:18]: 3.x is the latest "minor" version of python 3, currently, this is 3.10, but will change to 3.11 as soon as it's released.
This way, if a new version breaks stuff, it is noted right away.

I will remove 3.6, no need to support it with EOL exceeded.

**Review by fhs — APPROVED** [review]
Thanks!

**Pull request merged**